src/main/tmdb/partials/awesomeSearch/AwesomeSearchController.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * AwesomeSearchController provides controller support for inline searching
+ *
+ * @module tmdb.partials.search.AwesomeSearchController
+ *
+ * @requires angular
+ * @requires config
+ * @requires TMDBAPIService
+ *
+ * @author John Gomez <jgomez@alertlogic.com>
+ *
+ * @returns instance of the AwesomeSearchController
+ *
+ * @copyright Alert Logic, Inc 2016
+ *
+ */
+
+define( [ 'angular',
+          'config/config',
+          'tmdb/services/TMDBAPIService'],
+    function( angular, config, TMDBAPIService ) {
+        "use strict";
+
+        var AwesomeSearchController = function($scope, $rootScope, TMDBAPIService, $timeout, localStorage ) {
+            //Reference variables
+            var self = this;
+            var apiSearch = TMDBAPIService.Search();
+            var apiPerson = TMDBAPIService.Person();
+            var searchPromise;
+            self.timeWatcher = 500;
+            var limitName = 24;
+
+            var config  = angular.module("config");
+            var defaultImage = "http://www.lifeisajourney.org/wp-content/uploads/2013/03/Changed-Movie-Icon.png";
+
+            
+            var baseURL = "undefined baseURL";
+
+            $rootScope.$on( '$routeChangeSuccess', function() {
```

Comment:
This is perfect based on last week's material, but there are a few improvements that are important moving forward:

1.  The configuration retrieval/storage should be handled in a service, like the application state service we worked on today.  The $routeChangeSuccess event isn't necessary there, and we would only need to load the data from TMDBAPIService if we don't already have configuration data.

2.  Controllers (like AwesomeSearchResultsController) and views (like awesomeSearchResults.html) that show image references should use the image base URL to construct image references, using ng-src (https://docs.angularjs.org/api/ng/directive/ngSrc).